agent.go

```diff
@@ -11,27 +11,44 @@ import (
 
 	"github.com/GoogleCloudPlatform/kubectl-ai/gollm"
 	"github.com/GoogleCloudPlatform/kubectl-ai/pkg/journal"
+	"github.com/GoogleCloudPlatform/kubectl-ai/pkg/llmstrategy"
 	"github.com/GoogleCloudPlatform/kubectl-ai/pkg/ui"
 	"k8s.io/klog/v2"
 )
 
 // Agent knows how to execute a multi-step task. Goal is provided in the query argument.
 type Agent struct {
```

Comment:
another option to organize will be have different agent type structs, that all use common pieces to implement an agent type and top level `SuperAgent` is composed of sub-agents and in this case, only subagent type (ReAgent or ChatBasedAgent).

Reply:
I have no objection to refactoring the commonalities once we have the two types :-)